d2cli/main.go

```diff
@@ -749,26 +737,46 @@ func renderPDF(ctx context.Context, ms *xmain.State, plugin d2plugin.Plugin, opt
 		if err != nil {
 			return svg, err
 		}
-		err = doc.AddPDFPage(pngImg, currBoardPath, opts.ThemeID, rootFill, diagram.Shapes, int64(opts.Pad), viewboxX, viewboxY, pageMap)
+		err = doc.AddPDFPage(pngImg, boardPath, opts.ThemeID, rootFill, diagram.Shapes, int64(opts.Pad), viewboxX, viewboxY, pageMap)
 		if err != nil {
 			return svg, err
 		}
 	}
 
+	var prevBoardID string
+	var lastIndex int
+	if len(diagram.Layers) > 0 || len(diagram.Scenarios) > 0 || len(diagram.Steps) > 0 {
+		lastIndex = len(boardPath)
+		boardPath = append(boardPath, pdf.BoardTitle{})
```

Comment:
why are you doing this? 

in the loops below, why not just append a new board title? 

unless i'm missing something, this code is only confusing because you've made it that way. if you do it right it should be a lot clearer

Reply:
```
boardPath = [root]
layers = [layer1, layer2]

# first iter
boardPath = append(boardPath, dl)
# results in boardPath = [root, layer1]
renderPDF()

# second iter
boardPath = append(boardPath, dl)
# results in boardPath = [root, layer1, layer2], wrong path, should be [root, layer2]
renderPDF()
```

The same problem happens for scenarios and steps
After appending the layers, boardPath would be `[root, layers, ...]` and then scenarios would append there instead of having `[root, scenarios]`

Reply:
use `:=` then?

Reply:
yeah, I updated it..
Still, the below can happen

```
var boardPath []string

boardPath = append(boardPath, "a")
boardPath = append(boardPath, "b")
boardPath = append(boardPath, "c")

boardPath2 := append(boardPath, "d")
boardPath3 := append(boardPath, "e")

fmt.Printf("%v\n", boardPath)
fmt.Printf("%v\n", boardPath2)
fmt.Printf("%v\n", boardPath3)

// expected
// [a b c]
// [a b c d]
// [a b c e]
// actual
// [a b c]
// [a b c e]
// [a b c e]
```